src/pages/UserList/UserTable.js

```diff
@@ -48,7 +50,21 @@ const UserTable = ({
     if (!loading && users.length === 0) {
         return (
             <DataTableInfoWrapper columns={5}>
-                <p>{i18n.t('No results found')}</p>
+                <EmptyTableInfo
+                    header={i18n.t('No results found')}
+                    text={i18n.t(
+                        'There are no users to show here. You can add a new user below.'
+                    )}
+                    action={
```

Comment:
Ideally, there would be two cases:
- if table is empty because an active filter returns 0 results, show a "Clear filters" button.
- if table is empty because 0 results without a filter, show a "Add new user" button.

However, as I understood it, there wasn't a simple way to check for this (e.g. using `if (filters && users.length === 0)` or similar), because the filter component lives outside of this `UserTable` component. There is surely a way, but that was beyond the scope of these changes I thought.

If there's a simple way of implementing the two different cases though, that'd be great.

Reply:
Got you. If you want, I could make a ticket to update this after you merge this PR (I think it should be okay to pass information about whether there any filters down from the parent component).

There still, however, wouldn't be an easy way to know if the query is empty because of the filters or because there are no users (and will remain empty even if filters are removed) (as you could navigate to the user list with filters in the URL and hence we would not have any prior info about query results without filters), but in any case prompting to clear filters as the first step seems reasonable?

Reply:
> but in any case prompting to clear filters as the first step seems reasonable?

Yep, good point. I do think it'd be worth having the button as "Clear filters" if there's 0 results **and** a filter applied, so if that's possible then great!

Reply:
I agree with @cooper-joe that it would be nice to differentiate between these two cases. And agree with @tomzemp that it's probably cleanest to pass this information down from the parent component. But I actually think we don't need to add a separate ticket for this, since it should be fairly straightforward. I also tried to verify the followwing:
> There still, however, wouldn't be an easy way to know if the query is empty because of the filters or because there are no users (and will remain empty even if filters are removed) (as you could navigate to the user list with filters in the URL and hence we would not have any prior info about query results without filters)

But I don't think this is the case from what I saw.

As far as I can tell, the following would be needed:
- Add a `hasFilter` prop to the `UserTable` component
- In the `UserList` component, populate the `hasFilterProp` with the following expression:
   ```jsx
   hasFilterProp={!!query || !!inactiveMonths || !!invitationStatus || selfRegistered || organisationUnits.length > 0}
   ``` 
- Render a different action and text for the `EmptyTableInfo` based on the presence of `hasFilterProp`

And the other list components would need something similar. Extending the scope of this PR to include this seems like a good solution to me, but if you prefer to tackle this separately, that's also fine.

Reply:
I think it's worth handling this in a separate PR, perhaps writing up some guidelines of which actions we offer at in different circumstances.

Reply:
@tomzemp I think what you're saying is true in general. When there are 0 results and there is a filter, this could either be because there are no filter matches, or because there are no records at all. But I don't think this is a very likely scenario, and it would also be catered for pretty nicely by offering these two messages:
1. When there are 0 results and a filter is applied, we offer to clear the filter by clicking a button
2. If there are still zero results after clearing the filter, we offer to add a record by clicking a button

@cooper-joe if you prefer to tackle this separately, please create a JIRA issue and paste a link in a comment to this conversation.

Reply:
> @cooper-joe if you prefer to tackle this separately, please create a JIRA issue and paste a link in a comment to this conversation.

https://dhis2.atlassian.net/browse/DHIS2-14485